frontend/app/src/components/dashboard/DashboardAssetTable.vue

```diff
@@ -124,20 +144,30 @@ import {
   ref,
   toRefs
 } from '@vue/composition-api';
+import { DataTableHeader } from 'vuetify';
+import ShowedColumnsSelector from '@/components/dashboard/ShowedColumnsSelector.vue';
+import MenuTooltipButton from '@/components/helper/MenuTooltipButton.vue';
 import {
   setupAssetInfoRetrieval,
   setupExchangeRateGetter
 } from '@/composables/balances';
 import { currency } from '@/composables/session';
+import { setupSettings } from '@/composables/settings';
 import { totalNetWorthUsd } from '@/composables/statistics';
 import { aggregateTotal } from '@/filters';
 import i18n from '@/i18n';
 import { Nullable } from '@/types';
+import { DashboardTableType } from '@/types/frontend-settings';
+import { TableColumn } from '@/types/table-column';
 import { getSortItems } from '@/utils/assets';
 
-const tableHeaders = (totalNetWorthUsd: Ref<BigNumber>) =>
-  computed(() => {
-    return [
+const tableHeaders = (
+  totalNetWorthUsd: Ref<BigNumber>,
+  title: string,
+  showedColumns: TableColumn[]
```

Comment:
So the reason why this does not work is because `showedColumns` is not a reactive property but just a value.

`showedColumns` will keep the originally assigned value for the whole lifecycle of the [computed](https://v3.vuejs.org/api/computed-watch-api.html#computed) property.

To keep the computed property reactive what you have to do in this case is to pass the computed `Ref` _this means (`dashboardTablesShowedColumns`)_, to the `tableHeaders` function along with the `tableType` and `unref` _(this means calling: (`tableType.value`)_ only inside the `computed` closure and not inside the setup function.

_Sidenote: could we replace `showed` with `visible` e.g. `visibleColumns`_

In general if we want a computed property to react to data we have to pass a `ref`.

Let me know if this helps.

Reply:
Fixed